### PR TITLE
refactor: self-contain SSO and IDV components

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-2392](https://openedx.atlassian.net/browse/PROD-2392)
Separated IDVStatus and SSORecords components and moved their respective api calls, modals and states.

This PR does the following visual changes:

1. Changed the size of PageLoading from 50 to 20
2. Added a PageLoading component while it fetches the data from api as shown below:
Screenshot 2021-07-07 at 1 48 55 PM

3. Added AlertList to handle alerts inside the components as shown below:
Screenshot 2021-07-07 at 2 54 10 PM

(Coverage yet to be addressed)

### CODE REVIEW

File: src/users/IDVStatus.jsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import PropTypes from 'prop-types';
+import { Modal, Button } from '@edx/paragon';
+import Table from '../Table';
+import UserMessagesContext from '../userMessages/UserMessagesContext';
+import { getUserVerificationStatus } from './data/api';
+import { formatDate } from '../utils';
+import PageLoading from '../components/common/PageLoading';
+import AlertList from '../userMessages/AlertList';
+
+export default function IDVStatus({
```
Comment (DawoudSheraz) [inline src/users/IDVStatus.jsx]: please use full name(IdentityVerificationStatus)

File: src/users/SSORecords.jsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import PropTypes from 'prop-types';
+import { Modal, Button } from '@edx/paragon';
+import Table from '../Table';
+import UserMessagesContext from '../userMessages/UserMessagesContext';
+import { getSsoRecords } from './data/api';
+import { formatDate } from '../utils';
+import PageLoading from '../components/common/PageLoading';
+import AlertList from '../userMessages/AlertList';
+
+export default function SSORecords({
```
Comment (DawoudSheraz) [inline src/users/SSORecords.jsx]: Same here. SingleSignOnRecords

File: src/users/IDVStatus.jsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  const { add } = useContext(UserMessagesContext);
+  const [verificationData, setVerificationData] = useState(null);
+  const [detailIdvDataTitle, setDetailIdvDataTitle] = useState('');
+  const [detailIdvData, setDetailIdvData] = useState([]);
+  const [idvModalIsOpen, setIdvModalIsOpen] = useState(false);
+  useEffect(() => {
+    getUserVerificationStatus(username).then((result) => {
+      const camelCaseResult = camelCaseObject(result);
+      if (camelCaseResult.errors) {
+        camelCaseResult.errors.forEach(error => add(error));
```
Comment (DawoudSheraz) [inline src/users/IDVStatus.jsx]: clear any existing errors before adding new ones. Otherwise, it will provide an inconsistent user experience where they will be seeing any previous errors if present

File: src/users/IDVStatus.jsx (hunk 1 above)
Comment (Ali-D-Akbar) [reply src/users/IDVStatus.jsx]: done.

File: src/users/SSORecords.jsx (hunk 2 above)
Comment (Ali-D-Akbar) [reply src/users/SSORecords.jsx]: Done.

File: src/users/IDVStatus.jsx (hunk 3 above)
Comment (Ali-D-Akbar) [reply src/users/IDVStatus.jsx]: Done.

File: src/users/IdentityVerificationStatus.test.jsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+          topic: 'idvStatus',
+        },
+      ],
+    };
+    jest.spyOn(api, 'getUserVerificationStatus').mockImplementationOnce(() => Promise.resolve(idvErrors));
+    wrapper = mount(<IdentityVerificationStatusWrapper {...props} />);
+    await waitForComponentToPaint(wrapper);
+
+    const alert = wrapper.find('div.alert');
+    expect(alert.test).toEqual(idvErrors.errors.text);
```
Comment (azanbinzahid) [inline src/users/IdentityVerificationStatus.test.jsx]: what is `alert.test` ?

**Review by azanbinzahid — APPROVED** [review]
LGTM

File: src/users/IdentityVerificationStatus.test.jsx (hunk 4 above)
Comment (Ali-D-Akbar) [reply src/users/IdentityVerificationStatus.test.jsx]: Uh oh, this is a case of typo plus undefined === undefined. 
Lemme re-write these lines.

File: src/users/IdentityVerificationStatus.jsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+import AlertList from '../userMessages/AlertList';
+
+export default function IdentityVerificationStatus({
+  username,
+}) {
+  const { add, clear } = useContext(UserMessagesContext);
+  const [verificationData, setVerificationData] = useState(null);
+  const [detailIdvDataTitle, setDetailIdvDataTitle] = useState('');
+  const [detailIdvData, setDetailIdvData] = useState([]);
+  const [idvModalIsOpen, setIdvModalIsOpen] = useState(false);
```
Comment (DawoudSheraz) [inline src/users/IdentityVerificationStatus.jsx]: nit: isIdvModalOpen

File: src/users/SingleSignOnRecords.jsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+import { formatDate } from '../utils';
+import PageLoading from '../components/common/PageLoading';
+import AlertList from '../userMessages/AlertList';
+
+export default function SingleSignOnRecords({
+  username,
+}) {
+  const [extraSsoDataTitle, setSsoExtraDataTitle] = useState('');
+  const [ssoExtraData, setSsoExtraData] = useState([]);
+  const [ssoModalIsOpen, setSsoModalIsOpen] = useState(false);
```
Comment (DawoudSheraz) [inline src/users/SingleSignOnRecords.jsx]: nit: isSsoModalOpen

File: src/users/entitlements/Entitlements.test.jsx [hunk 7]
```diff
@@ -5,13 +5,13 @@ import { waitForComponentToPaint } from '../../setupTest';
...
-import UserMessageProvider from '../../userMessages/UserMessagesProvider';
+import UserMessagesProvider from '../../userMessages/UserMessagesProvider';
 import * as api from '../data/api';
 
 const EntitlementsPageWrapper = (props) => (
-  <UserMessageProvider>
+  <UserMessagesProvider>
     <Entitlements {...props} />
-  </UserMessageProvider>
+  </UserMessagesProvider>
```
Comment (DawoudSheraz) [inline src/users/entitlements/Entitlements.test.jsx:14]: Why is this changed?

**Review by DawoudSheraz — APPROVED** [review]
a few nits to address before merging.

File: src/users/entitlements/Entitlements.test.jsx (hunk 7 above)
Comment (Ali-D-Akbar) [reply src/users/entitlements/Entitlements.test.jsx:14]: It was a typo from the previous changes.
